lib/appmap/service/config_analyzer.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+module AppMap
+  module Service
+    class ConfigAnalyzer
+      attr_reader :config_error
+
+      def initialize(config_file)
+        @config_file = config_file
+        @config = load_config
+      end
+
+      def app_name
+        @config.to_h[:name] if is_present?
+      end
+
+      def is_present?
+        File.exist?(@config_file)
+      end
+
+      def is_valid?
+        is_present? && @config.to_h.key?(:name) && @config.to_h.key?(:packages)
+      end
+
+      private
+
+      def load_config
+        AppMap::Config.load_from_file @config_file if is_present?
```

Comment:
Here we can implement the different error conditions that might occur. So, first check if the file is present. Then try and parse it as YAML. If it's valid YAML, then call `AppMap::Config.load`, and rescue any errors.